tests/test_fv3run.py

```diff
@@ -0,0 +1,252 @@
+        config_file.flush()
+        subprocess.check_call(
+            ['fv3run', config_file.name, outdir, '--runfile', MOCK_RUNSCRIPT]
+        )
+
+
+def docker_run(config_dict, outdir):
+    with tempfile.NamedTemporaryFile(mode='w') as config_file:
+        config_file.write(yaml.dump(config_dict))
+        config_file.flush()
+        try:
+            subprocess.check_call(
+                ['fv3run', config_file.name, outdir, '--runfile', MOCK_RUNSCRIPT,
+                 '--dockerimage', DOCKER_IMAGE_NAME]
+            )
+        except subprocess.CalledProcessError as err:
+            if os.path.isfile(os.path.join(outdir, 'stdout.log')):
+                with open(os.path.join(outdir, 'stdout.log')) as stdout:
+                    print(f'STDOUT: {stdout.read()}')
+            if os.path.isfile(os.path.join(outdir, 'stderr.log')):
+                with open(os.path.join(outdir, 'stderr.log')) as stderr:
+                    print(f'STDERR: {stderr.read()}')
+            raise err
+
+
+def config_dict_module_run(config_dict, outdir):
+    fv3config.run(config_dict, outdir, runfile=MOCK_RUNSCRIPT)
+
+
+def config_dict_filename_run(config_dict, outdir):
+    with tempfile.NamedTemporaryFile(mode='w') as config_file:
+        config_file.write(yaml.dump(config_dict))
+        config_file.flush()
+        fv3config.run(config_file.name, outdir, runfile=MOCK_RUNSCRIPT)
+
+
+def count_executed_ranks(rundir):
+    filenames = os.listdir(rundir)
+    total = 0
+    while f'rank_{total}' in filenames:
+        total += 1
+    return total
+
+
+@contextlib.contextmanager
+def cleaned_up_directory(dirname):
+    try:
+        yield
+    finally:
+        if os.path.isdir(dirname):
+            shutil.rmtree(dirname)
+
+
+def check_run_directory(dirname):
+    filenames = os.listdir(dirname)
+    assert 'stdout.log' in filenames
+    assert 'stderr.log' in filenames
+    assert count_executed_ranks(dirname) == 6
+    assert 'input.nml' in filenames
+    assert 'INPUT' in filenames
+
+
+@pytest.mark.parametrize(
+    "runner",
+    [config_dict_module_run, config_dict_filename_run]
+)
+def test_fv3run_with_mocked_subprocess(runner):
+    if runner == docker_run and not DOCKER_ENABLED:
+        pytest.skip(f'docker or docker image {DOCKER_IMAGE_NAME} is not available')
+    fv3config.ensure_data_is_downloaded()
+    outdir = os.path.join(TEST_DIR, 'outdir')
+
+    with unittest.mock.patch("subprocess.check_call") as mock, cleaned_up_directory(outdir):
+        runner(fv3config.get_default_config(), outdir)
+        assert mock.called
+        assert mock.call_args[0] == ([
+            'mpirun', '-n', '6', 'python3', '-m', 'mpi4py', 'mock_runscript.py'],)
+        config = yaml.safe_load(open(os.path.join(outdir, 'fv3config.yml'), 'r'))
+        assert config == fv3config.get_default_config()
+
+
+def test_fv3run_docker():
+    """End-to-end test of running a mock runscript inside a docker container.
+    """
+    if not DOCKER_ENABLED:
```

Comment:
Not necessary to change, but `pytest` has a [skipif decorator][1] that does this.

[1]:  http://doc.pytest.org/en/latest/skipping.html#id1

Reply:
Updated. This was left over from when the tests were bundled in a way that didn't allow using the decorator.